lib/omniauth/strategy.rb

```diff
@@ -347,8 +347,8 @@ def extra
     def auth_hash
       hash = AuthHash.new(:provider => name, :uid => uid)
       hash.info = info unless skip_info?
-      hash.credentials = credentials if credentials
-      hash.extra = extra if extra
+      (credentials_data = credentials) && hash.credentials = credentials_data
+      (extra_data = extra) && hash.extra = extra_data
```

Comment:
Using local variables (instead of instance variables) for memoization to minimize unintended side effect. If you find these one liners to be unreadable, I'm happy to separate assignments into their own lines @BobbyMcWho